### DISCUSSION TITLE
Jetty 12 Upgrade

### DISCUSSION THREAD

**Discussion by stevenschlansker** [resteasy/resteasy#4310] [General]
@jamezp , we are looking to upgrade to Jetty 12. Right now resteasy jetty client supports Jetty 11, which has reached "End Of Community Support" (not quite yet end of support, but soon)

The simplest path is to just update to Jetty 12 at some convenient resteasy minor version. I don't know how many resteasy-client-jetty users are out there, but this would be a breaking change for them.

Another option is to copy the whole client library as is to a -jetty12 artifact, fix it up, and deprecate the existing on.

If we want to support multiple Jetty versions at once, I could try to pull apart an abstract superclass - additional complexity, but easier to change Jetty versions.

How would the resteasy project want to approach a Jetty 12 transition? Thanks!

**Comment by jamezp** [comment]
@stevenschlansker I definitely don't want to add a jetty-12 module. I think we did that too much in the past and IMO it becomes unwieldy.

TBH my preferred approach would be to pull it out of the resteasy project and place it in it's own project. The reason is I don't know when I'd do a new minor.

That said, we could make 7.0.0 use Jetty 12.

My personal option would be to deprecate it in the resteasy project though and create a new resteasy-jetty project with it's own release cycle.

**Reply by stevenschlansker** [reply]
Where are you imagining this project would live? A new repo in the Resteasy org?

**Reply by jamezp** [reply]
Yes, just something like https://github.com/resteasy/resteasy-jetty.

**Reply by stevenschlansker** [reply]
Ok, works for me. I will work to prepare the Jetty 12 ready version of the client. Would you mind setting up the project at your convenience, or how should I more formally request this? Thanks!

**Reply by jamezp** [reply]
No need for a formal request IMO. I'll setup the repository and just copy over what we have. I should have time today. I'll note I'm going on PTO starting tomorrow and all of next week.

One thing to consider would be to create a 1.0.0.Final release of what we have now targeting Jetty 11 with package name changes. Then, work on a 2.0. targeting Jetty 12.

I think for the package name, since we have to change it so we don't have package splits, would be to use `dev.resteasy.jetty` as the base package. Then the groupId for the Maven module would change to `dev.resteasy.jetty`, but we could keep the artifacts the same probably.

I'll also file a JIRA to deprecate it in RESTEasy and update the documentation to reflect the changes.

Does that sound reasonable?

**Reply by stevenschlansker** [reply]
Great thank you, all sounds good to me. Enjoy your well deserved time off :)

**Reply by jamezp** [reply]
FWIW. RESTEasy 7.0.0.Alpha3 is already using Jetty 12. Since I copied from the main branch, it looks like Jetty 12 is already there :) I'll be pushing this very soon as it's almost done. If it helps, I could immediately cut say a 1.0.0.Alpha1 so you have something to test with.

**Reply by stevenschlansker** [reply]
Oh, very cool! Maybe I don't need to write much code at all. We'd love to test it out if you have time to make a release. (We can always build a local one ourselves if not.)

**Reply by jamezp** [reply]
Okay, project created https://github.com/resteasy/resteasy-jetty and a 1.0.0.Alpha1 was released. It should be available in Maven Central soon, maybe 30 minutes or so.

**Reply by jamezp** [reply]
Just to keep the link between this, I've filed an issue to deprecate the client in RESTEasy itself https://issues.redhat.com/browse/RESTEASY-3541.

**Reply by stevenschlansker** [reply]
Thanks, we're using the alpha release successfully so far in our test branch for a week or two now.
